src/platform/nodejs/lib/util/crypto.ts

```diff
@@ -0,0 +1,279 @@
+
+  /**
+   * Internal: calculate the padded length of a given plaintext
+   * using PKCS5.
+   * @param plaintextLength
+   * @return
+   */
+  function getPaddedLength(plaintextLength: number) {
+    return (plaintextLength + DEFAULT_BLOCKLENGTH) & -DEFAULT_BLOCKLENGTH;
+  }
+
+  /**
+   * Internal: checks that the cipherParams are a valid combination. Currently
+   * just checks that the calculated keyLength is a valid one for aes-cbc
+   */
+  function validateCipherParams(params: API.Types.CipherParams) {
+    if (params.algorithm === 'aes' && params.mode === 'cbc') {
+      if (params.keyLength === 128 || params.keyLength === 256) {
+        return;
+      }
+      throw new Error(
+        'Unsupported key length ' +
+          params.keyLength +
+          ' for aes-cbc encryption. Encryption key must be 128 or 256 bits (16 or 32 ASCII characters)'
+      );
+    }
+  }
+
+  function normaliseBase64(string: string) {
+    /* url-safe base64 strings use _ and - instread of / and + */
+    return string.replace('_', '/').replace('-', '+');
+  }
+
+  /**
+   * Internal: obtain the pkcs5 padding string for a given padded length;
+   */
+  function filledBuffer(length: number, value: number) {
+    var result = Buffer.alloc(length);
+    result.fill(value);
+    return result;
+  }
+  var pkcs5Padding = [filledBuffer(16, 16)];
+  for (var i = 1; i <= 16; i++) pkcs5Padding.push(filledBuffer(i, i));
+
+  /**
+   * Internal: convert a binary string to Buffer (for node 0.8.x)
+   * @param bufferOrString
+   * @returns {Buffer}
+   */
+  function toBuffer(bufferOrString: Buffer | string) {
+    return typeof bufferOrString == 'string' ? Buffer.from(bufferOrString, 'binary') : bufferOrString;
+  }
+
+  /**
+   * A class encapsulating the client-specifiable parameters for
+   * the cipher.
+   *
+   * algorithm is the name of the algorithm in the default system provider,
+   * or the lower-cased version of it; eg "aes" or "AES".
+   *
+   * Clients may instance a CipherParams directly and populate it, or may
+   * query the implementation to obtain a default system CipherParams.
+   */
+  class CipherParams implements API.Types.CipherParams {
+    algorithm: string;
+    keyLength: number;
+    mode: string;
+    key: NodeCipherKey;
+    iv: unknown;
+
+    constructor(algorithm: string, keyLength: number, mode: string, key: NodeCipherKey) {
+      this.algorithm = algorithm;
+      this.keyLength = keyLength;
+      this.mode = mode;
+      this.key = key;
+      this.iv = null;
+    }
+  }
+
+  function isInstCipherParams(
+    params: API.Types.CipherParams | API.Types.CipherParamOptions
+  ): params is API.Types.CipherParams {
+    /* In node, can't use instanceof CipherParams due to the vm context problem (see
+     * https://github.com/nwjs/nw.js/wiki/Differences-of-JavaScript-contexts).
+     * So just test for presence of all necessary attributes */
+    return params.algorithm && params.key && params.keyLength && params.mode ? true : false;
```

Comment:
very much a personal preference but i prefer:
```suggestion
    return !!(params.algorithm && params.key && params.keyLength && params.mode);
```

Reply:
Have changed it 👍